v3/src/components/tool-shelf/tiles-list-button.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react"
+import { observer } from "mobx-react-lite"
+import { Menu, MenuButton, MenuItem, MenuList } from "@chakra-ui/react"
+import OptionsIcon from "../../assets/icons/icon-options.svg"
+import { useDocumentContent } from "../../hooks/use-document-content"
+import { uiState } from "../../models/ui-state"
+import { IFreeTileLayout } from "../../models/document/free-tile-row"
+import { t } from "../../utilities/translation/translate"
+import { kRightButtonBackground, ToolShelfButtonTag } from "./tool-shelf-button"
+import GraphIcon from "../../assets/icons/icon-graph.svg"
+import TableIcon from "../../assets/icons/icon-table.svg"
+import SliderIcon from "../../assets/icons/icon-slider.svg"
+import MapIcon from "../../assets/icons/icon-map.svg"
+import CalculatorIcon from "../../assets/icons/icon-calc.svg"
+import GuideIcon from "../../assets/icons/icon-guide.svg"
+import WebViewIcon from "../../assets/icons/icon-media-tool.svg"
+
+import "./tool-shelf.scss"
+
+export const TilesListShelfButton = observer(function TilesListShelfButton() {
+  const documentContent = useDocumentContent()
+  const tilesArr: any[] = []
+
+  documentContent?.tileMap.forEach((tile) => {
+    tilesArr.push(tile)
+  })
+
+  const tileTypeIconMap = [{type: "Graph", icon: <GraphIcon className="tile-list-icon"/>},
+                            {type: "CaseTable", icon: <TableIcon className="tile-list-icon"/>},
+                            {type: "CodapSlider", icon: <SliderIcon className="tile-list-icon"/>},
+                            {type: "Map", icon: <MapIcon className="tile-list-icon"/>},
+                            {type: "Calculator", icon: <CalculatorIcon className="tile-list-icon"/>},
+                            {type: "Guide", icon: <GuideIcon className="tile-list-icon"/>},
+                            {type: "CodapWebView", icon: <WebViewIcon className="tile-list-icon"/>}
+                          ]
+
+  const handleSelectTile = (tileId: string) => {
+    uiState.setFocusedTile(tileId)
+    const tileRow = documentContent?.findRowContainingTile(tileId)
+    const tileLayout = tileRow?.getTileLayout(tileId) as IFreeTileLayout
+    tileLayout.setMinimized(false)
```

Comment:
I think we only use `FreeTileLayout` in Codap, but it would be safer to test that `tileLayout` is of type `FreeTileLayout`, and then only call `setMinimized` if it is.